component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -1045,11 +1045,7 @@ protected void processEmailOTPFlow(HttpServletRequest request, HttpServletRespon
                 Boolean.parseBoolean(emailOTPParameters.get(EmailOTPAuthenticatorConstants.SHOW_AUTH_FAILURE_REASON));
         AuthenticatedUser authenticatedUser = getAuthenticatedUser(context);
         try {
-            if (authenticatedUser == null) {
-                throw new AuthenticationFailedException("Authentication failed!. Cannot proceed further without " +
-                        "identifying the user");
-            }
-            if (isLocalUser(context) && EmailOTPUtils.isAccountLocked(authenticatedUser)) {
+            if (isLocalUser(context) && authenticatedUser != null && EmailOTPUtils.isAccountLocked(authenticatedUser)) {
```

Comment:
what if the authenticated user is null ? How are we handling that scenario ?

Reply:
Updated the integration test to handle it https://github.com/wso2/product-is/pull/16609

Reply:
The authenticated user can be null if the first factor is local authenticator, hence the null check cannot be used

Reply:
authenticatedUser cannot be null if isLocalUser() is ture. therfore no need of a null check here